### PR TITLE
Setup mongoose

### PR DESCRIPTION
**Pull request by razorblade446**
* Services, Schemas and models added
* Controller added
* Need to test why had to use forwardRef for BookService
* Controller setup for books
* tsconfig update

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -232,4 +232,6 @@ modules.xml
 *.ipr
 
 
-# End of https://www.gitignore.io/api/osx,node,intellij,webstorm+all
\ No newline at end of file
+# End of https://www.gitignore.io/api/osx,node,intellij,webstorm+all
+
+src/config.json
```
Comment (salorrego) [inline .gitignore]: EOF

**Review by salorrego — COMMENTED** [review]
te falto un EOF en el gitignore

File: .gitignore (hunk 1 above)
Comment (razorblade446) [reply .gitignore]: Done!

**Review by salorrego — APPROVED** [review]
LGTM

**Pull request merged**